CMakeLists.txt

```diff
@@ -46,6 +46,22 @@ if(OSQUERY_ENABLE_CLANG_TIDY)
   endif()
 endif()
 
+# We need to figure out the architecture and set the normalized variable
+# before any targets are created
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
+  # Windows x86_64
+  add_compile_definitions(__x86_64__)
```

Comment:
This should be moved inside `cmake/flags.cmake`
```
if("${TARGET_PROCESSOR}" STREQUAL "x86_64")
  list(APPEND
    __x86_64__
  )
endif()
````

Reply:
It's only for windows. I'll skip it for now, we can decide what to do for Windows on ARM64 when the time comes